onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -65,6 +95,25 @@ void CpuToCpuMemCpy(void* dst, const void* src, size_t num_bytes) {
   memcpy(dst, src, num_bytes);
 }
 
+OrtMemoryInfo GetMemoryInfoPerDeviceType(const OrtDevice& ort_device) {
+  OrtMemoryInfo mem_info;
+  if (ort_device.Type() == OrtDevice::CPU) {
+    mem_info = GetAllocator()->Info();
+  }
+#if USE_CUDA
+  else if (ort_device.Type() == OrtDevice::GPU) {
+    if (!IsCudaDeviceIdValid(logging::LoggingManager::DefaultLogger(), ort_device.Id())) {
```

Comment:
Should this support ROCM as well?

Reply:
I do not have expertise on ROCM.

Reply:
Response from Suffian:
I'm not yet sure if the call to IsCudaDeviceIdValid applicable to both CUDA and ROCm.
Turning on for ROCm could be a separate PR.

Reply:
I'd expect IsRocmDeviceIdValid should be called